menu/actions/PhyMenuActions.c

```diff
@@ -0,0 +1,60 @@
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "boot.h"
+
+int IsSerialEnabled(void)
+{
+	int result;
+
+	IoOutputByte(0x2e, 0x55);	// Enter Configuration
+
+	// Select serial device
+	IoOutputByte(0x2e, 0x07);
+	IoOutputByte(0x2f, 0x04);
+
+	// Check whether device is enabled
+	IoOutputByte(0x2e, 0x30);
+	result = IoInputByte(0x2f);
+
+	IoOutputByte(0x2e, 0xAA);	// Exit Configuration
+
+	if (result > 1) return 0;
+	return result;
+}
+
+void SetSerialEnabled(void *menuItemText)
+{
+	char *text = (char *)menuItemText;
+
+	IoOutputByte(0x2e, 0x55);	// Enter Configuration
+
+	// Select serial device
+	IoOutputByte(0x2e, 0x07);
+	IoOutputByte(0x2f, 0x04);
+
+	// Enable or Disable device
+	IoOutputByte(0x2e, 0x30);
+	IoOutputByte(0x2f, strstr(text, "Enabled") ? 0x00 : 0x01);
```

Comment:
This looks very dirty. Why is this being done?

Is this function called by some config script or something?
Why does it have to search for a case-sensitive string instead of receiving a `bool enabled`?

Reply:
The menu have text only structure, there is no place for boolean flags. So comparison either should be done here or somewhere else in the callers.

Reply:
Fair enough, I see the problem.

I'd still suggest something like:
```c
int current_state = IsSerialEnabled();
int wanted_state = current_state ? 0x00 : 0x01;
```
(Must be done before entering configuration state in this function)